File: lemur-libs/lemur-testkit/src/main/java/io/github/malczuuu/lemur/testkit/annotation/ContainerTest.java
package io.github.malczuuu.lemur.testkit.annotation;

import static io.github.malczuuu.lemur.testkit.TestkitTags.TESTCONTAINERS;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.junit.jupiter.api.Tag;

/**
 * Meta-annotation for test fields that require Testcontainers infrastructure. Used for filtering
 * out integration tests for a quick build.
 */
@Tag(TESTCONTAINERS)
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface ContainerTest {}

File: lemur-libs/lemur-testkit/src/main/kotlin/io/github/malczuuu/lemur/testkit/annotation/ContainerTest.kt
package io.github.malczuuu.lemur.testkit.annotation

import io.github.malczuuu.lemur.testkit.TestkitTags.TESTCONTAINERS
import org.junit.jupiter.api.Tag

@Tag(TESTCONTAINERS)
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class ContainerTest
